test/FunctionalTests/Microsoft.OData.Edm.Tests/Csdl/CsdlReaderJsonTests.cs

```diff
@@ -1858,6 +1858,41 @@ public void ParsingPropertyWithUntypedTypeAsPropertyTypeInJsonWorks()
             Assert.Equal("Collection(Edm.Untyped)", dataProperty.Type.FullName());
         }
 
+        [Fact]
+        public void ReadJsonKeyAlias()
+        {
+            var csdl = @"{
+""$Version"": ""4.0"",
+""$EntityContainer"": ""NS1.Container"",
+  ""NS1"": {
+    ""Product"": {
+      ""$Kind"": ""EntityType"",
+      ""$Key"": [
+        {
+          ""ExampleObjectName"": ""NS1.ExampleComplexObject/ExampleObject""
+        }
+      ],
+      ""ExampleObject"": {
+        ""$Type"": ""NS1.ExampleComplexObject""

```

Comment:
The `NS1.ExampleComplexObject` type does not appear to be defined in this CSDL schema.

Reply:
added it.